File: frontend/angular/src/app/core/dtos/supervisor-dashboard/supervisor-dashboard.dto.ts
import { SupervisorDashboardOperatorDto } from './supervisor-dashboard-operator.dto';
import { SupervisorDashboardOrderDto } from './supervisor-dashboard-order.dto';

export interface SupervisorDashboardDto {
  machinesRunning: number;
  operatorsActive: number;
  lateJobs: number;
  operators: SupervisorDashboardOperatorDto[];
  orders: SupervisorDashboardOrderDto[];
}

File: backend/CncApp/CncApp.Application/Dtos/SupervisorDashboard/SupervisorDashboardDto.cs
namespace CncApp.Application.Dtos.SupervisorDashboard;

public class SupervisorDashboardDto
{
    public int MachinesRunning { get; set; }
    public int OperatorsActive { get; set; }
    public int LateJobs { get; set; }
    public List<SupervisorDashboardOperatorDto> Operators { get; set; } = new();
}
